### PR TITLE
[luci-interpreter]Add InstanceNorm on luci-interpreter

### PR DESCRIPTION
**Pull request by struss**
This commit add InstanceNorm on luci-interpreter.

ONE-DCO-1.0-Signed-off-by: KiDeuk Bang <rrstrous@nate.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/InstanceNorm.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      evalFloat();
+      break;
+    default:
+      throw std::runtime_error("Unsupported type.");
+  }
+}
+
+void InstanceNorm::evalFloat() const
+{
+  float activation_min, activation_max;
```
Comment (binarman) [inline compiler/luci-interpreter/src/kernels/InstanceNorm.cpp:58]: Just to be sure.
Did you take any of these code from tensorflow?
If yes add TF authors to license, please.

**Review by binarman — APPROVED** [review]
LGTM

File: compiler/luci-interpreter/src/kernels/InstanceNorm.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    : KernelWithParams<InstanceNormParams>({input, gamma, beta}, {output}, params)
+{
+}
+
+void InstanceNorm::configure()
+{
+  LUCI_INTERPRETER_CHECK(input()->shape().num_dims() == 4);
+  LUCI_INTERPRETER_CHECK(input()->element_type() == output()->element_type());
+  LUCI_INTERPRETER_CHECK(gamma()->element_type() == DataType::FLOAT32);
+  LUCI_INTERPRETER_CHECK(beta()->element_type() == DataType::FLOAT32);
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/InstanceNorm.cpp:40]: These checks are valid only in fp32 model.

For future quantized kernels, let's change the code as below.

```suggestion
  LUCI_INTERPRETER_CHECK(gamma()->element_type() == input()->element_type());
  LUCI_INTERPRETER_CHECK(beta()->element_type() == input()->element_type());
```

File: compiler/luci-interpreter/src/kernels/InstanceNorm.cpp [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+}
+
+void InstanceNorm::evalFloat() const
+{
+  float activation_min, activation_max;
+  calculateActivationRange(params().activation, &activation_min, &activation_max);
+  auto input_shape = getTensorShape(input());
+  auto output_shape = getTensorShape(output());
+  const int32_t batches = tflite::MatchingDim(input_shape, 0, output_shape, 0);
+  const int32_t heights = tflite::MatchingDim(input_shape, 1, output_shape, 2);
```
Comment (jinevening) [inline compiler/luci-interpreter/src/kernels/InstanceNorm.cpp:63]: Maybe this?

```suggestion
  const int32_t heights = tflite::MatchingDim(input_shape, 1, output_shape, 1);
```

File: compiler/luci-interpreter/src/kernels/InstanceNorm.cpp (hunk 1 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/InstanceNorm.cpp:58]: Oh, This comes from OneRT code runtime code base.

File: compiler/luci-interpreter/src/kernels/InstanceNorm.cpp (hunk 2 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/InstanceNorm.cpp:40]: Thanks for pointing out!

File: compiler/luci-interpreter/src/kernels/InstanceNorm.cpp (hunk 3 above)
Comment (struss) [reply compiler/luci-interpreter/src/kernels/InstanceNorm.cpp:63]: Sorry I will fix it.

**Review by jinevening — APPROVED** [review]
LGTM

**Comment by jinevening** [comment]
@seanshpark Can you merge this PR? @binarman already approved before, and today is Russian holiday (Unity day).

**Pull request merged**